### PR TITLE
lars cookbook

### PR DESCRIPTION
**Pull request by Saurabh7**

### CODE REVIEW

**Comment by karlnapf** [comment]
This needs integration testing data

File: doc/cookbook/source/examples/regression/least_angle_regression.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
+=======================
+Least Angle Regression
+=======================
+
+Least Angle Regression (LARS) is an algorithm used to fit a linear regression model. LARS is simliar to forward stagewise regression but less greedy. Instead of including variables at each step, the estimated parameters are increased in a direction equiangular to each one's correlations with the residual. LARS can be used to solve LASSO, which is basically L1-regularized least square regression.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/regression/least_angle_regression.rst]: "a sparse linear regression model"

Comment (karlnapf) [inline doc/cookbook/source/examples/regression/least_angle_regression.rst]: remove "basically"

File: doc/cookbook/source/examples/regression/least_angle_regression.rst [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+Imagine we have files with training and test data. We create `CDenseFeatures` (here 64 bit floats aka RealFeatures) and :sgclass:`CRegressionLabels` as
+
+.. sgexample:: least_angle_regression.sg:create_features
+
+To normalize and center the features, we create an instance of preprocessors :sgclass:`CPruneVarSubMean` and :sgclass:`CNormOne` and apply it on the feature matrices.
+
+.. sgexample:: least_angle_regression:preprocess_features
+
+We create an instance of :sgclass:`CLeastAngleRegression` by passing False to disable LASSO solution, setting the penalty :math:`\lambda` for l1 norm and setting training data and labels.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/regression/least_angle_regression.rst]: whats this false thing?
What is it for?

File: doc/cookbook/source/examples/regression/least_angle_regression.rst [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+.. sgexample:: least_angle_regression:create_instance
+
+Then we train the regression model and apply it to test data to get the predicted :sgclass:`CRegressionLabels` .
+
+.. sgexample:: linear_ridge_regression.sg:train_and_apply
+
+After training, we can extract :math:`{\bf w}`.
+
+.. sgexample:: linear_ridge_regression.sg:extract_w
```
Comment (karlnapf) [inline doc/cookbook/source/examples/regression/least_angle_regression.rst]: we also want the regularisation path extraction

Comment (Saurabh7) [inline doc/cookbook/source/examples/regression/least_angle_regression.rst]: Its a std vector  of vectors  right now, can be copied into and sgmatrix and extracted. Will do that.

**Comment by Saurabh7** [comment]
@karlnapf  this looks ok ?

**Comment by karlnapf** [comment]
Looks good to me.
What we need is

 * squash commits
 * send a data PR with integration testing data (before merging this one)
 * once that is merged, also update the data submodule hash in here so that integration tests are executed

**Comment by Saurabh7** [comment]
Yes, I have linked the data pr. https://github.com/shogun-toolbox/shogun-data/pull/122

**Comment by karlnapf** [comment]
I just merged it, so only squash and data update

**Comment by karlnapf** [comment]
Hey @Saurabh7 would you mind squashing these? Then we can merge

**Comment by Saurabh7** [comment]
Yep, squashed and rebased.

**Comment by karlnapf** [comment]
Another candidate for  #3580

**Pull request merged**